src/main/java/in/ashwanthkumar/gocd/github/provider/gitlab/GitLabProvider.java

```diff
@@ -0,0 +1,155 @@
+import org.gitlab4j.api.models.Author;
+import org.gitlab4j.api.models.MergeRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+public class GitLabProvider implements Provider {
+    private static final Logger LOG = LoggerFactory.getLogger(GitLabProvider.class);
+    public static final String REF_SPEC = "+refs/merge-requests/*/head:refs/remotes/origin/merge-requests/*";
+    public static final String REF_PATTERN = "refs/remotes/origin/merge-requests/";
+
+    @Override
+    public GoPluginIdentifier getPluginId() {
+        return new GoPluginIdentifier("gitlab.pr", Arrays.asList("1.0"));
+    }
+
+    @Override
+    public String getName() {
+        return "Gitlab";
+    }
+
+    @Override
+    public void addConfigData(GitConfig gitConfig) {
+        try {
+            Properties props = GitLabUtils.readPropertyFile();
+            if (StringUtil.isEmpty(gitConfig.getUsername())) {
+                gitConfig.setUsername(props.getProperty("login"));
+            } if (StringUtil.isEmpty(gitConfig.getPassword())) {
+                gitConfig.setPassword(props.getProperty("password"));
+            }
+        } catch (IOException e) {
+            // ignore
+        }
+    }
+
+    @Override
+    public boolean isValidURL(String url) {
+        return new URLUtils().isValidURL(url);
+    }
+
+    @Override
+    public void checkConnection(GitConfig gitConfig) {
+        try {
+            Objects.requireNonNull(loginWith(gitConfig)).getProjectApi().getProjects();
+        } catch (Exception e) {
+            throw new RuntimeException(String.format("check connection failed. %s", e.getMessage()), e);
+        }
+    }
+
+    @Override
+    public String getRefSpec() {
+        return REF_SPEC;
+    }
+
+    @Override
+    public String getRefPattern() {
+        return REF_PATTERN;
+    }
+
+    @Override
+    public void populateRevisionData(GitConfig gitConfig, String prId, String prSHA, Map<String, String> data) {
+        data.put("PR_ID", prId);
+
+        PullRequestStatus prStatus = null;
+        boolean isDisabled = System.getProperty("go.plugin.gitlab.pr.populate-details", "Y").equals("N");
+        LOG.debug("Populating PR details is disabled");
+        if (!isDisabled) {
+            prStatus = getPullRequestStatus(gitConfig, prId, prSHA);
+        }
+
+        if (prStatus != null) {
+            data.put("PR_BRANCH", String.valueOf(prStatus.getPrBranch()));
+            data.put("TARGET_BRANCH", String.valueOf(prStatus.getToBranch()));
+            data.put("PR_URL", String.valueOf(prStatus.getUrl()));
+            data.put("PR_AUTHOR", prStatus.getAuthor());
+            data.put("PR_AUTHOR_EMAIL", prStatus.getAuthorEmail());
+            data.put("PR_DESCRIPTION", prStatus.getDescription());
+            data.put("PR_TITLE", prStatus.getTitle());
```

Comment:
Does all these properties work with Gitlab as well?

Reply:
I was able to access these properties from `MergeRequest` model and that's how I set it up in the transform method.